Core/Src/main.c

```diff
@@ -74,7 +78,46 @@ int main(void)
   HAL_Init();
 
   /* USER CODE BEGIN Init */
-
+  	  // pow(2,10)-1 because of 10 bits, considering pot response 0-1023
+	  const int DUTY_CYCLE_COUNTS_5_PERCENT = 3200;
+
+	  /* I set the prescaler to 14, and the total count period
+		   * to 64000.
+		   * To convert pot response (0-1023) to duty cycle counts
+		   * (3200-6400), the values must correspond in this fashion:
+		   *
+		   * 0 => 3200, and 1023 => 6400
+		   *
+		   * This should follow a standard formula. If 0 corresponds to
+		   * 3200, that means the formula to convert pot response to
+		   * counts must include a scalar addition of 3200. Followign a
+		   * mx + b fashion, the multiplier must be as follows:
+		   *
+		   * 6400 = m(1023) + 3200
+		   * or that m = 3.128f*/
+  	  const double SCALING_FACTOR = 3.128;
+
+  	  //Timeout for spi transmit receive function in ms
+  	  const uint32_t SPI_TIMEOUT = 250;
+  	  const uint8_t NUM_BITS_TO_SEND = 3;
+  	  //Bit mask as we only need the last 2 bits in the second returned byte
+  	  const uint8_t BIT_MASK = 0x00000011;
+
+  	  /*MOSI data, according to figure 6.1 MCU transmitted data
+  	   * first byte: 00000001
+  	   * second byte: 10000000: 128: 0x80
+  	   * third byte: doesn't matter, using 00000000
+  	   */
+  	  uint8_t tx_data[NUM_BITS_TO_SEND] = {0x1, 0x80, 0x0};
+
+  	  //MISO data
+  	  uint8_t rx_data[NUM_BITS_TO_SEND] = {0x0};
+  	  uint8_t second_byte = 0;
+  	  uint8_t third_byte = 0;
+
+  	  uint16_t pot_response = 0;
+  	  int counts = 0;
```

Comment:
Keep in mind, in embedded systems it is always best to use a data type that most closely resembles the data that it will represent. So if counts will never be below 0, you should use uint. Also in general, it is best to use a uint data types with a specific value if you know what it will be less than, for ex, uint8_t, uint16_t reason being it is explicit for the reader and then you know exactly how much memory this data is taking up.